### PR TITLE
UIIN-2754: Add the ability to update ownership of items

### PR DESCRIPTION
**Pull request by OleksandrHladchenko1**
## Purpose

 * Be able to move items from one FOLIO tenant to another

## Approach

* Display **Update ownership** button for user with multiple affiliations for Item linked to Shared Instance and when user has **Inventory: Update ownership** permission.
* Transformed `ItemView` component into the functional one in order to use hooks such as `useHoldingMutation` and `useUpdateOwnership`
* Added `tenant` parameter to `useHoldingMutation` hook in order to have the ability to create holding for another tenant.
* Added `UpdateItemOwnershipModal` component that contains two fields: `Selection` component with affiliations (except current and central one) and `FieldHolding` component that allows us to choose holding for the chosen tenant OR create a new holding, based on the location.
* Added `path` parameter to `useUpdateOwnership` hook in order to use it for updating ownership of both holdings and items.
* Added unit tests

## Refs

[UIIN-2754](https://folio-org.atlassian.net/browse/UIIN-2754)

## Screenshots

https://github.com/user-attachments/assets/e98610a2-4235-40f2-adab-f6e1a87a6fe0

### CODE REVIEW

**Review by BogdanDenis — CHANGES_REQUESTED** [review]
Please increase code coverage to 80%, especially for ItemView component because it has a lot of changes

File: src/components/UpdateItemOwnershipModal/UpdateItemOwnershipModal.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+      <Form
+        onSubmit={onSubmit}
+        render={({ values }) => (
+          <>
+            <Field name="affiliation">
+              {fieldProps => (
+                <Selection
+                  {...fieldProps.input}
+                  label={<FormattedMessage id="ui-inventory.affiliation" />}
+                  placeholder="Select affiliation"
```
Comment (mariia-aloshyna) [inline src/components/UpdateItemOwnershipModal/UpdateItemOwnershipModal.js:96]: let's make it translatable

File: src/components/UpdateItemOwnershipModal/UpdateItemOwnershipModal.test.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  instanceId: 'instanceId',
+  tenantsList: [{ id: 'university', name: 'University' }],
+};
+
+const renderUpdateItemOwnershipModal = () => renderWithIntl(
+  <UpdateItemOwnershipModal {...defaultProps} />,
+  translationsProperties
+);
+
+describe('UpdateItemOwnershipModal', () => {
```
Comment (mariia-aloshyna) [inline src/components/UpdateItemOwnershipModal/UpdateItemOwnershipModal.test.js:39]: add test for axe, please

**Comment by OleksandrHladchenko1** [comment]
> Please increase code coverage to 80%, especially for ItemView component because it has a lot of changes

Hi @BogdanDenis I increased test coverage for ItemView component. Please review this PR one more time.
image

File: src/views/ItemView.js [hunk 3]
```diff
@@ -853,6 +856,7 @@ const ItemView = props => {
       return (
         <Link
           to={`/inventory/view/${x.briefInstance?.id}`}
+          aria-label={`Instance HRID - ${x.briefInstance?.id}`}
```
Comment (BogdanDenis) [inline src/views/ItemView.js:859]: Please add this to translations

**Review by BogdanDenis — CHANGES_REQUESTED** [review]

File: src/views/ItemView.js (hunk 3 above)
Comment (OleksandrHladchenko1) [reply src/views/ItemView.js]: Done!

File: src/views/ItemView.js [hunk 4]
```diff
@@ -597,1138 +484,1342 @@ class ItemView extends React.Component {
...
+            <Col xs={1}>
+              <KeyValue
+                label={<FormattedMessage id="ui-inventory.staffOnly" />}
+                value={get(item, ['notes'], []).map((note, j) => {
+                  if (note.itemNoteTypeId === noteType.id) {
+                    return <div key={j}>{note.staffOnly ? 'Yes' : 'No'}</div>;
+                  }
+
+                  return null;
+                })}
```
Comment (mariia-aloshyna) [inline src/views/ItemView.js:635]: I would suggest moving this and the ones below to a variable and using `useMemo` so as not to re-render values each time.

File: src/views/ItemView.js [hunk 5]
```diff
@@ -597,1138 +484,1342 @@ class ItemView extends React.Component {
...
+      <FormattedMessage id="stripes-core.button.back" />
+    </Button>
+  );
+
+  const administrativeData = {
+    discoverySuppress: get(instance, 'discoverySuppress', '-'),
+    hrid: get(item, 'hrid', '-'),
+    barcode: get(item, 'barcode', '-'),
+    accessionNumber: get(item, 'accessionNumber', '-'),
+    identifier: get(item, 'itemIdentifier', '-'),
```
Comment (mariia-aloshyna) [inline src/views/ItemView.js:714]: use `NoValue` instead of `-`

File: src/views/ItemView.js [hunk 6]
```diff
@@ -597,1138 +484,1342 @@ class ItemView extends React.Component {
...
+    'URI': x => {
+      const uri = x?.uri;
+
+      return uri
+        ? (
+          <a
+            href={uri}
+            rel="noreferrer noopener"
+            target="_blank"
+            style={wrappingCell}
```
Comment (mariia-aloshyna) [inline src/views/ItemView.js:842]: can `TextLink` be used here?

**Review by mariia-aloshyna — APPROVED** [review]

**Pull request merged**